Core/Src/monitor.c

```diff
@@ -14,185 +14,46 @@
-		temp_msg.data[3] = (humidity >> 8) & 0xFF;
-
-		/* Send CAN message */
-		if (queue_can_msg(temp_msg)) {
-			fault_data.diag = "Failed to send CAN message";
-			queue_fault(&fault_data);
-		}
-
-		/* Yield to other tasks */
-		osDelay(TEMP_SENS_SAMPLE_DELAY);
-	}
+	pedals_t temp;
+	osMutexAcquire(pedals_mutex, osWaitForever);
+	memcpy(&temp, &pedal_data, sizeof(pedals_t));
+	osMutexRelease(pedals_mutex);
+	return temp;
 }
 
-osThreadId_t pedals_monitor_handle;
-const osThreadAttr_t pedals_monitor_attributes = {
-	.name = "PedalMonitor",
-	.stack_size = 128 * 8,
-	.priority = (osPriority_t)osPriorityRealtime1,
-};
-
-void eval_pedal_fault(uint16_t accel_1, uint16_t accel_2,
-		      nertimer_t *diff_timer, nertimer_t *sc_timer,
-		      nertimer_t *oc_timer, fault_data_t *fault_data)
+bool get_brake_state()
 {
-	/* Fault - open circuit (Max ADC value + some a lil bit) */
-	if ((accel_1 > (MAX_ADC_VAL_12b - 20) ||
-	     accel_2 > (MAX_ADC_VAL_12b - 20)) &&
-	    is_timer_active(oc_timer)) {
-		if (is_timer_expired(oc_timer)) {
-			if ((accel_1 == MAX_ADC_VAL_12b ||
-			     accel_2 == MAX_ADC_VAL_12b)) {
-				fault_data->diag =
-					"Pedal open circuit fault - max acceleration value ";
-				queue_fault(fault_data);
-			}
-		}
-	} else if ((accel_1 > (MAX_ADC_VAL_12b - 20) ||
-		    accel_2 > (MAX_ADC_VAL_12b - 20)) &&
-		   !is_timer_active(oc_timer)) {
-		start_timer(oc_timer, PEDAL_FAULT_TIME);
-	} else {
-		cancel_timer(oc_timer);
-	}
-
-	/* Fault - short circuit */
-	if ((accel_1 < 500 || accel_2 < 500) && is_timer_active(sc_timer)) {
-		if (is_timer_expired(sc_timer)) {
-			if ((accel_1 < 500 || accel_2 < 500)) {
-				fault_data->diag =
-					"Pedal short circuit fault - no acceleration value ";
-				queue_fault(fault_data);
-			}
-		}
-	} else if ((accel_1 < 500 || accel_2 < 500) &&
-		   !is_timer_active(sc_timer)) {
-		start_timer(sc_timer, PEDAL_FAULT_TIME);
-	} else {
-		cancel_timer(sc_timer);
-	}
-
-	/* Normalize pedal values */
-	uint16_t accel_1_norm =
-		accel_1 - ACCEL1_OFFSET <= 0 ?
-			0 :
-			(uint16_t)(accel_1 - ACCEL1_OFFSET) * 100 /
-				(ACCEL1_MAX_VAL - ACCEL1_OFFSET);
-	uint16_t accel_2_norm =
-		accel_2 - ACCEL2_OFFSET <= 0 ?
-			0 :
-			(uint16_t)(accel_2 - ACCEL2_OFFSET) * 100 /
-				(ACCEL2_MAX_VAL - ACCEL2_OFFSET);
-
-	/* Fault - difference between pedal sensing values */
-	if ((abs(accel_1_norm - accel_2_norm) > PEDAL_DIFF_THRESH) &&
-	    is_timer_active(diff_timer)) {
-		/* starting diff timer */
-		if (is_timer_expired(diff_timer)) {
-			if ((abs(accel_1_norm - accel_2_norm) >
-			     PEDAL_DIFF_THRESH)) {
-				fault_data->diag =
-					"Pedal fault - pedal values are too different ";
-				queue_fault(fault_data);
-			}
-		}
-	} else if ((abs(accel_1_norm - accel_2_norm) > PEDAL_DIFF_THRESH) &&
-		   !is_timer_active(diff_timer)) {
-		start_timer(diff_timer, PEDAL_FAULT_TIME);
-	} else {
-		cancel_timer(diff_timer);
-	}
+	return get_pedal_data().brake_value > PEDAL_BRAKE_THRESH;
```

Comment:
very minor performance thing but if we acquire the mutex and check the data we can avoid the extra memcpy.  prob not necessary but a small performance improvment as using a bool in memory instead of the complete pedal data.

Reply:
I had that but I didnt like the extra mutex and static variable that came with it.

Reply:
I thought it's already in a mutex?

Reply:
this needs a mutex anyway - never want to directly read a value that another thread can change.. This needs the safety of aquriing the mutex before reading, unless im missing something